### ISSUE TITLE
Expand Default Value of `constantWaiverParentToken` in MagicNumberCheck with All Operators

### ISSUE THREAD

**Issue by mahfouz72** [checkstyle/checkstyle#15985]
Check documentation: https://checkstyle.org/checks/coding/magicnumber.html#MagicNumber

**Rationale of the check:** 
> Checks that there are no ["magic numbers"](https://en.wikipedia.org/wiki/Magic_number_%28programming%29) where a magic number is a numeric literal that is not defined as a constant.

> constantWaiverParentToken: Specify tokens that are allowed in the AST path from the number literal to the enclosing constant definition.

The rationale of the check is to violate all magic numbers that are not defined as a constant and `constantWaiverParentToken`
specify tokens allowed in enclosing constant definition.

I suggest expanding the allow list with all operators (bitwise, comparison, arithmetic, logical, etc..) because:

- The property concerns only constant fields, which should be always ignored.
- To maintain consistency suggest ignoring all operators by default.
- This would align with the rationale that numbers inside constant fields are inherently non-magic numbers.

```Java
public class Example {
    public static final int CONSTANT = 7 * 6; // Consistency in ignoring all operators
    public static final boolean CONDITION = 7 >> 6; // Consistency in ignoring all operators
}
```

This simplifies check configuration for users and ensures that all constant fields are treated uniformly, regardless of the operators used.
Also, users have the option to override this property with any tokens that they need. with common sense, I would say that all operators should be included here.

**Comment by romani** [comment]
As this property only for constants (final fields) I am ok to add all operators to default. If we miss some, in update, it is ok, user will find and report it, we will do follow up update.

If user wants to be strict in his code, he needs to override default with operators he likes.

@mahfouz72, please restore your PR with link to this issue.

**Comment by mahfouz72** [comment]
@romani done

**Comment by nrmancuso** [comment]
Closed via #15957
